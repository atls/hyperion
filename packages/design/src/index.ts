export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
