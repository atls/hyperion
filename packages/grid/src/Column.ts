export const Column = (props: any) => null
