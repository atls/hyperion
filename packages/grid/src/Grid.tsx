import useComponentSize   from '@rehooks/component-size'
import React, { useRef }  from 'react'

import { Column, Layout } from '@atlantis-lab/layout'

import { Body }           from './body'
import { Header }         from './header'

export const Grid = ({
  children,
  data = [],
  rowHeight = 40,
  loading,
  loadMore,
  order,
  onOrder,
}) => {
  const ref = useRef(null)
  const { width } = useComponentSize(ref)

  return (
    <Column ref={ref}>
      <Layout>
        <Header columns={children} order={order} onOrder={onOrder} />
      </Layout>
      <Layout height='100%' flexGrow={1}>
        {width && (
          <Body
            data={data}
            columns={children}
            rowHeight={rowHeight}
            rowWidth={width}
            loading={loading}
            loadMore={loadMore}
          />
        )}
      </Layout>
    </Column>
  )
}
