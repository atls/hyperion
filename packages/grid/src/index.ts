export * from './Column'
export * from './Grid'
