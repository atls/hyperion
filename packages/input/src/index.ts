export * from './Textarea'
export * from './Input'
export * from './Checkbox'
