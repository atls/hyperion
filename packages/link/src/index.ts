export * from './Link'
export * from './NextLink'
export * from './RouteLink'
