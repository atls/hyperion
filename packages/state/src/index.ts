export * from './withState'
export * from './useActive'
export * from './useHover'
export * from './useFocus'
