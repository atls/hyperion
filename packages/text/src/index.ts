export * from './Text'
export * from './Space'
export * from './SearchResult'
