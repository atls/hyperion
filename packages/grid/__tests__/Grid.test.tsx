import React            from 'react'
import renderer         from 'react-test-renderer'

import { Column, Grid } from '../src/index'

it('Grid should render correctly', () => {
  const cols = ['Column', 'Another']

  const tree = renderer
    .create(
      <Grid data={[]} order={{ by: 'desc' }} loading={false} loadMore={false} onOrder={false}>
        {cols.map((i, idx) => (
          <Column header={`${i} ${idx}`} orderBy='desc' basis='100px' />
        ))}
      </Grid>,
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
