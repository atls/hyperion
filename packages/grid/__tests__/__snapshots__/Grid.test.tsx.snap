// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render correctly 1`] = `
.emotion-7 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  border-bottom: 0.5px solid #1f2639;
  box-sizing: border-box;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Lato",sans-serif;
  padding: 10px 0px;
  overflow: hidden;
  box-sizing: border-box;
  font-size: 14px;
  color: #6d84a9;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding-left: 10px;
  outline: none;
}

.emotion-0 {
  margin-left: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-0 svg:last-child {
  position: relative;
  left: -3px;
}

.emotion-6 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="emotion-7"
  display="flex"
  height="100%"
  width="100%"
>
  <div
    className="emotion-5"
    display="flex"
  >
    <div
      className="emotion-4"
    >
      <div
        className="emotion-1"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "flexBasis": "100px",
            "flexShrink": 0,
          }
        }
      >
        Column 0
        <span
          className="emotion-0"
        >
          <svg
            fill="none"
            height="1em"
            stroke="#ffffff"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 19V6M5 12l7-7 7 7"
            />
          </svg>
          <svg
            fill="none"
            height="1em"
            stroke="#ffffff"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 5v13M5 12l7 7 7-7"
            />
          </svg>
        </span>
      </div>
      <div
        className="emotion-1"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "flexBasis": "100px",
            "flexShrink": 0,
          }
        }
      >
        Another 1
        <span
          className="emotion-0"
        >
          <svg
            fill="none"
            height="1em"
            stroke="#ffffff"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 19V6M5 12l7-7 7 7"
            />
          </svg>
          <svg
            fill="none"
            height="1em"
            stroke="#ffffff"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 5v13M5 12l7 7 7-7"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <div
    className="emotion-6"
    display="flex"
    height="100%"
  >
    0
  </div>
</div>
`;
